toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -79,6 +79,37 @@ func New() *RpmRepoCloner {
 	return &RpmRepoCloner{}
 }
 
+// ConstrcuctClonerWithNetwork constructs a new RpmRepoCloner with optional network access.
+//   - destinationDir is the directory to save RPMs
+//   - tmpDir is the directory to create a chroot
+//   - workerTar is the path to the worker tar used to seed the chroot
+//   - existingRpmsDir is the directory with prebuilt RPMs
+//   - prebuiltRpmsDir is the directory with toolchain RPMs
+//   - tlsCert is the path to the TLS certificate, "" if not needed
+//   - tlsKey is the path to the TLS key, "" if not needed
+//   - usePreviewRepo if set, the upstream preview repository will be used.
+//   - disableUpstreamRepos if set, the upstream repositories will not be used.
+//   - repoDefinitions is a list of repo files to use
+func ConstrcuctClonerWithNetwork(destinationDir, tmpDir, workerTar, existingRpmsDir, toolchainRpmsDir, tlsCert, tlsKey string, usePreviewRepo, disableUpstreamRepos bool, repoDefinitions []string) (r *RpmRepoCloner, err error) {
+	timestamp.StartEvent("initialize and configure cloner", nil)
+	timestamp.StopEvent(nil) // initialize and configure cloner
```

Comment:
This looks fishy. Why are we stopping the event immediatelly after we started it?

Reply:
Ah, missing a defer. Good catch